posts/Martin/2015-07-13-introduction-to-hibernate-search-genericjpa.adoc

```diff
@@ -0,0 +1,460 @@
+	}
+}
+----
+
+=== What did we do?
+- Class-Level:
+	* added @Indexed annotation
+		** This is needed for Hibernate Search to recognize this entity-class as a Top-Level index class.
+	* added @InIndex annotation
+		** This is a special annotation needed by Hibernate Search GenericJPA on every 
+			entity that somehow is found in any index. Just put it there and you’ll be fine
+- Field-Level:
+	* added @DocumentId/@Field on name
+		** Hibernate Search needs to know which field is used to identify this entity in the index.
+			this produces a field with the name “id” in the index.
+			We also want to store the name into a field called “name”.
+	* added @Field on author
+		** apart from searching for the name we also want to be able to search for
+			the author of the book. this is stored into the field called “author” in the index.
+			
+== Starting up the engine
+
+As Hibernate Search GenericJPA is not integrated into Hibernate ORM we have to 
+manually start everything up. But this is not hard, at all:
+
+[source,java]
+----
+//first, let's start up the basic JPA persistence.
+EntityManagerFactory emf = 
+  Persistence.createEntityManagerFactory( "EclipseLink_MySQL" );
+EntityManager em = emf.createEntityManager();
+ 
+//now, let's get the Properties
+Properties hsearchConfiguration = "...";
+//start our JPASearchFactoryController
+JPASearchFactoryController searchController =
+  Setup.createSearchFactory( emf, hsearchConfiguration );
+----
+
+You may have noticed loading a properties file in this snippet.
+This is the configuration needed for Hibernate Search. Let’s take a look at it next.
+
+[source]
+----
+# we use are using MySQL. this is needed to 
+# create the triggers for updating
+org.hibernate.search.genericjpa.searchfactory.triggerSource=\
+org.hibernate.search.genericjpa.db.events.triggers.MySQLTriggerSQLStringSource
+ 
+# we have a searchfactory that is supposed 
+#to get it's updates from SQL
+org.hibernate.search.genericjpa.searchfactory.type=sql
+ 
+# how many updates should be processed at once?
+# this number is quite conservative
+org.hibernate.search.genericjpa.searchfactory.batchSizeForUpdates=10
+ 
+# what delay do we want between checking for updates?
+org.hibernate.search.genericjpa.searchfactory.updateDelay=25
+ 
+# we are in a Java SE environment in this example.
+# if you specify 'true', you will have to add an extra property
+# to lookup the JTA mechanism. This is however not needed
+# if you are managing your Transactions by yourself
+# org.hibernate.search.genericjpa.searchfactory.transactionManagerProvider.jndi=\
+# java:jboss/TransactionManager
+org.hibernate.search.genericjpa.searchfactory.useJTATransactions=false
+ 
+# for this example we store our directory in ram.
+hibernate.search.default.directory_provider=ram
+ 
+# we could add extra properties as specified in the Hibernate Search 
+# documentation (http://docs.jboss.org/hibernate/search/5.3/reference/en-US/html_single/#_configuration) next
```

Comment:
Better link to http://docs.jboss.org/hibernate/stable/search/reference/. And make it nicely rendered with a text.